api/src/main/java/org/openmrs/api/db/hibernate/HibernateFormDAO.java

```diff
@@ -475,17 +475,17 @@ private Criteria getFormCriteria(String partialName, Boolean published, Collecti
 				allFormFieldIds.add(ff.getFormFieldId());
 			}
 			DetachedCriteria subquery = DetachedCriteria.forClass(FormField.class, "ff");
-			subquery.setProjection(Projections.count("ff.formFieldId"));
-			subquery.add(Restrictions.eqProperty("ff.form", "form"));
-			subquery.add(Restrictions.in("ff.formFieldId", allFormFieldIds));
-			
+			subquery.setProjection(Projections.count("ff.formFieldId")).
+			add(Property.forName("ff.form.formId").eqProperty("f.formId")).
+			add(Restrictions.in("ff.formFieldId", allFormFieldIds));
 			crit.add(Subqueries.eq((long) containingAllFormFields.size(), subquery));
+			
 		}
 		
 		// get all forms (dupes included) that have this field on them
 		if (!fields.isEmpty()) {
 			Criteria crit2 = crit.createCriteria("formFields", "ff");
-			crit2.add(Restrictions.eqProperty("ff.form.formId", "form.formId"));
+			crit2.add(Restrictions.eqProperty("ff.form.formId", "f.formId"));
```

Comment:
Do you have a unit test for this change?

Reply:
Well I have looked into this but this seems to be way too much hibernate. I expect the 2nd property of the above line to be the `id` of the current form. But surprisingly if you use `ff.form.formId`, it works then `f.formId` it also works and then `form.formId` it also works.

Please see : https://gist.github.com/samuelmale/1a52375a67e1ac0e47e347c3c3956404